addons/website/models/website.py

```diff
@@ -808,54 +808,42 @@ def search_url_dependencies(self, res_ids, res_model):
         dependencies = {}
         current_website = self.get_current_website()
 
-        # search for views containing the url, including the ones linked to a
-        # website.page
-        domain = []
-        urls = []
+        # Prepare what's needed to later generate the URL search domain for the
+        # given records
+        search_criteria = []
         for record in self.env[res_model].browse([int(res_id) for res_id in res_ids]):
             website = record.website_id or current_website
             url = getattr(record, 'website_url', False) or record.url
-            urls.append(url)
-            domain.append(AND([
-                [('arch_db', 'ilike', url)],
-                website.website_domain()
-            ]))
-        views = self.env['ir.ui.view'].search(OR(domain))
-
-        page_views = views.filtered('page_ids')
-        views -= page_views
-
-        if views:
-            view_key = _('Template') if len(views) == 1 else _('Templates')
-            dependencies[view_key] = [{
-                'content': Markup(_('Template <b>%s (id:%s)</b> contains a link to this page')) % (view.key or view.name, view.id),
-                'item': _('%s (id:%s)') % (view.key or view.name, view.id),
-                'link': '/web#id=%s&view_type=form&model=ir.ui.view' % view.id,
-            } for view in views]
-
-        if page_views:
-            page_key = _('Page') if len(page_views) == 1 else _('Pages')
-            dependencies[page_key] = [{
-                'content': Markup(_("Page <b>%s</b> contains a link to this page")) % page.url,
-                'item': page.name,
-                'link': page.url,
-            } for page in page_views.page_ids]
-
-        # search for menu containing the url
-        domain = []
-        for url in urls:
-            domain.append(AND([
-                [('url', 'ilike', url)],
-                website.website_domain(),
-            ]))
-        menus = self.env['website.menu'].search(OR(domain))
-        if menus:
-            menu_key = _('Menu') if len(menus) == 1 else _('Menus')
-            dependencies[menu_key] = [{
-                'content': Markup(_('This page is in the menu <b>%s</b>')) % menu.name,
-                'link': '/web#id=%s&view_type=form&model=website.menu' % menu.id,
-                'item': menu.name,
-            } for menu in menus]
+            search_criteria.append((url, website.website_domain()))
+
+        # Search the URL in every relevant field
+        html_fields_attributes = self._get_html_fields_attributes() + [
```

Comment:
Did not remember we already had that in the codebase (thought it was only upgrade repo), nice :+1: